### PR TITLE
Insufficient Gas

### PR DESCRIPTION
**Pull request by greg-schrammel**
Fixes APP-1665

## What changed (plus any additional context for devs)

adds a label "Insufficient Gas" for when the user has the funds but don't have enough gas tokens to execute the transaction

### CODE REVIEW

File: src/__swaps__/screens/Swap/providers/SyncSwapStateAndSharedValues.tsx [hunk 1]
```diff
@@ -128,7 +127,12 @@ export function SyncGasStateToSharedValues() {
 
   useEffect(() => {
     hasEnoughFundsForGas.value = undefined;
-    if (!gasSettings || !estimatedGasLimit || !quote || 'error' in quote || !userNativeNetworkAsset) return;
+    if (!gasSettings || !estimatedGasLimit || !quote || 'error' in quote || isLoadingNativeNetworkAsset) return;
+
+    if (!userNativeNetworkAsset) {
+      hasEnoughFundsForGas.value = false;
+      return;
+    }
```
Comment (greg-schrammel) [inline src/__swaps__/screens/Swap/providers/SyncSwapStateAndSharedValues.tsx:135]: tested with an account that didn't have any gas token, found out we have to differentiate between loading the `userNativeNetworkAsset` and not having it :)

File: src/languages/en_US.json [hunk 2]
```diff
@@ -1978,6 +1978,7 @@
         "swapping": "Swapping",
         "select_token": "Select Token",
         "insufficient_funds": "Insufficient Funds",
+        "insufficient_gas": "Insufficient Gas",
```
Comment (walmat) [inline src/languages/en_US.json:1981]: I'm not sure this label is clear to the user, however I don't have a better alternative off the top of my head. To me this reads as though we're failing to estimate gas or something. Insufficient funds for gas is better but too wordy. Will need to think on this.

Comment (greg-schrammel) [reply src/languages/en_US.json:1981]: another option I can think is `Insufficient <native>` like `Insufficient ETH`, `Insufficient BNB`

Comment (BrodyHughes) [reply src/languages/en_US.json:1981]: insufficient <native> i think is better tbh and more clear on what the issue is

**Review by BrodyHughes — APPROVED** [review]

File: bitrise.yml
Comment (walmat) [inline bitrise.yml:1]: rm this file

File: src/languages/en_US.json [hunk 3]
```diff
@@ -1978,6 +1978,7 @@
         "swapping": "Swapping",
         "select_token": "Select Token",
         "insufficient_funds": "Insufficient Funds",
+        "insufficient_gas": "Insufficient",
```
Comment (walmat) [inline src/languages/en_US.json:1981]:
```suggestion
        "insufficient": "Insufficient",
```

File: src/__swaps__/screens/Swap/providers/swap-provider.tsx [hunk 4]
```diff
@@ -606,6 +608,7 @@ export const SwapProvider = ({ children }: SwapProviderProps) => {
   }, []);
 
   const hasEnoughFundsForGas = useSharedValue<boolean | undefined>(undefined);
+  const gasAssetSymbol = useSharedValue<string>('Gas');
```
Comment (walmat) [inline src/__swaps__/screens/Swap/providers/swap-provider.tsx:611]: i think we probably want this to initially be based of the `internalSelectedInputAsset.value`, right?

Comment (walmat) [reply src/__swaps__/screens/Swap/providers/swap-provider.tsx:611]: I also think we could just name this `nativeAssetForChainSymbol`

File: src/__swaps__/screens/Swap/providers/swap-provider.tsx [hunk 5]
```diff
@@ -62,6 +62,7 @@ const review = i18n.t(i18n.l.swap.actions.review);
 const fetchingPrices = i18n.t(i18n.l.swap.actions.fetching_prices);
 const selectToken = i18n.t(i18n.l.swap.actions.select_token);
 const insufficientFunds = i18n.t(i18n.l.swap.actions.insufficient_funds);
+const insufficientGas = i18n.t(i18n.l.swap.actions.insufficient_gas);
```
Comment (walmat) [inline src/__swaps__/screens/Swap/providers/swap-provider.tsx:65]:
```suggestion
const insufficient = i18n.t(i18n.l.swap.actions.insufficient);
```

File: src/__swaps__/screens/Swap/providers/swap-provider.tsx [hunk 6]
```diff
@@ -667,7 +670,10 @@ export const SwapProvider = ({ children }: SwapProviderProps) => {
     }
 
     if (!hasEnoughFundsForGas.value) {
-      return { label: insufficientFunds, disabled: true };
+      return {
+        label: `${insufficientGas} ${gasAssetSymbol.value}`,
```
Comment (walmat) [inline src/__swaps__/screens/Swap/providers/swap-provider.tsx:674]:
```suggestion
        label: `${insufficient} ${gasAssetSymbol.value}`,
```

File: src/__swaps__/screens/Swap/providers/SyncSwapStateAndSharedValues.tsx [hunk 7]
```diff
@@ -128,7 +129,15 @@ export function SyncGasStateToSharedValues() {
...
-    if (!gasSettings || !estimatedGasLimit || !quote || 'error' in quote || !userNativeNetworkAsset) return;
+    if (!gasSettings || !estimatedGasLimit || !quote || 'error' in quote || isLoadingNativeNetworkAsset) return;
+
+    if (!userNativeNetworkAsset) {
+      hasEnoughFundsForGas.value = false;
+      const network = getNetworkFromChainId(chainId);
+      const { nativeCurrency } = getNetworkObj(network);
+      gasAssetSymbol.value = nativeCurrency.symbol;
+      return;
+    }
```
Comment (walmat) [inline src/__swaps__/screens/Swap/providers/SyncSwapStateAndSharedValues.tsx:140]: I'll reiterate what I put in DMs here for visibility:

Instead of reacting here in a useEffect we should only be updating the `gasAssetSymbol` when previousChainId !== currenctChainId (based off input currency).

File: src/__swaps__/screens/Swap/providers/swap-provider.tsx [hunk 8]
```diff
@@ -667,7 +668,11 @@ export const SwapProvider = ({ children }: SwapProviderProps) => {
     }
 
     if (!hasEnoughFundsForGas.value) {
-      return { label: insufficientFunds, disabled: true };
+      const nativeCurrency = RainbowNetworkByChainId[sellAsset?.chainId]?.nativeCurrency;
+      return {
+        label: `${insufficient} ${nativeCurrency?.symbol || 'Gas'}`,
+        disabled: true,
+      };
```
Comment (walmat) [inline src/__swaps__/screens/Swap/providers/swap-provider.tsx:675]: small nitpick.
```suggestion
      const nativeCurrency = RainbowNetworkByChainId[sellAsset?.chainId || ChainId.mainnet].nativeCurrency;
      return {
        label: `${insufficient} ${nativeCurrency.symbol}`,
        disabled: true,
      };
```

**Review by walmat — APPROVED** [review]
